code/modules/mob/living/carbon/xenomorph/castes/shrike/abilities_shrike.dm

```diff
@@ -344,44 +344,91 @@
-	desc = "Throw a gravity grenades thats sucks everyone and everything in a radius inward."
-	plasma_cost = 500
+
+// ***************************************
+// *********** Psychic Vortex
+// ***************************************
+#define VORTEX_RANGE 3
+#define VORTEX_PULL_WINDUP_TIME 2 SECONDS
+#define VORTEX_PUSH_WINDUP_TIME 1 SECONDS
+/datum/action/xeno_action/activable/psychic_vortex
+	name = "Pyschic vortex"
+	action_icon_state = "vortex"
+	desc = "Channel a sizable vortex of psychic energy, drawing in nearby enemies."
+	ability_name = "Psychic vortex"
+	plasma_cost = 600
+	cooldown_timer = 3 MINUTES
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
 	keybinding_signals = list(
-		KEYBINDING_NORMAL = COMSIG_XENOABILITY_GRAV_NADE,
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PSYCHIC_VORTEX,
 	)
-	cooldown_timer = 1 MINUTES
 
-/datum/action/xeno_action/activable/gravity_grenade/use_ability(atom/A)
-	var/turf/T = get_turf(owner)
+/datum/action/xeno_action/activable/psychic_vortex/on_cooldown_finish()
+	to_chat(owner, span_notice("Our mind is ready to unleash another chaotic vortex of energy."))
+	return ..()
+
+/datum/action/xeno_action/activable/psychic_vortex/use_ability(atom/target)
 	succeed_activate()
 	add_cooldown()
-	var/obj/item/explosive/grenade/gravity/nade = new(T)
-	nade.throw_at(A, 5, 1, owner, TRUE)
-	nade.activate(owner)
-
-	owner.visible_message(span_warning("[owner] vomits up a roaring fleshy lump and throws it at [A]!"), span_warning("We vomit up a roaring fleshy lump and throws it at [A]!"))
-
-
-/obj/item/explosive/grenade/gravity
-	name = "gravity grenade"
-	desc = "A fleshy mass that seems way too heavy for its size. It seems to be vibrating."
-	arm_sound = 'sound/voice/predalien_roar.ogg'
-	greyscale_colors = "#3aaacc"
-	greyscale_config = /datum/greyscale_config/xenogrenade
-	det_time = 20
-
-/obj/item/explosive/grenade/gravity/prime()
-	new /obj/effect/overlay/temp/emp_pulse(loc)
-	playsound(loc, 'sound/effects/EMPulse.ogg', 50)
-	for(var/atom/movable/victim in view(3, loc))//yes this throws EVERYONE
+	addtimer(CALLBACK(owner, TYPE_PROC_REF(/mob, update_icons)), 1 SECONDS)
+	var/mob/living/carbon/xenomorph/xeno = owner
+	owner.icon_state = "[xeno.xeno_caste.caste_name][xeno.is_a_rouny ? " rouny" : ""] Screeching"
+	if(target) // Keybind use doesn't have a target
+		owner.face_atom(target)
+	ADD_TRAIT(owner, TRAIT_IMMOBILE, VORTEX_ABILITY_TRAIT)
+	if(do_after(owner, VORTEX_PULL_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		vortex_pull()
+	if(do_after(owner, VORTEX_PUSH_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		vortex_push()
+	if(do_after(owner, VORTEX_PULL_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		vortex_pull()
+	finish_charging()
+	return fail_activate()
+
+/datum/action/xeno_action/activable/psychic_vortex/proc/vortex_pull()
+	playsound(owner, 'sound/effects/seedling_chargeup.ogg', 60)
+	for(var/atom/movable/victim AS in range(VORTEX_RANGE, owner.loc))
+		if(isarea(victim))
+			continue
```

Comment:
why?

Reply:
https://cdn.discordapp.com/attachments/500801682058379265/1117829973370228787/image.png
if I don't check for area, this will happen later on.

Reply:
dont cast this as movable but atom instead then since range returns turfs and areas